### ISSUE TITLE
New note creates a duplicate of current note

### ISSUE THREAD

**Issue by HaleemDream** [notable/notable#219]
Hey, not sure if this is how it is intended to work. When making a new note sometimes it creates an 'untitled' empty note (expected), other times, it creates a copy of the note (currently highlighted). Not sure if I'm doing something wrong but I have a clip to demo.

https://i.imgur.com/0QAEXBY.gif

**Comment by fabiospampinato** [comment]
This is definitely a bug, thanks for reporting it!

**Comment by jinjie** [comment]
Hint: Only happens when the editor is in editing mode.

Did a dirty trick where I toggle to read mode when creating a new note and it works.

```
  new = async () => {
    // Toggle out of edit first
    await this.ctx.editor.toggleEditing ( false );
```

**Comment by fabiospampinato** [comment]
@jinjie very useful, thanks for looking into this!

**Comment by RedRoxProjects** [comment]
Good to see this is being flagged - experienced the same issue today when creating notes in edit mode. 

Would be great if the default behaviour was as suggested - i.e.  toggle off edit mode (before note creation) if new note button is clicked whilst in edit mode. 

Loving the app so far! Really helpful for giving some structure to my PhD note taking! 🙌 ✨

**Comment by fabiospampinato** [comment]
> Would be great if the default behaviour was as suggested - i.e. toggle off edit mode (before note creation) if new note button is clicked whilst in edit mode.

That's an option but other people would like to pretty much always be in edit mode. Perhaps we could support both use cases, but for know let's just concentrate on fixing the bug.

**Comment by mapclyps** [comment]
for me a new note is created correctly even in edit mode if I use the shortcut for creating a new note using `CTRL+N` (or Note-> New Note) instead of the + button next to the search.

**Comment by fabiospampinato** [comment]
Working on fixing this right now, it's pretty weird that when using the shortcut for creating new notes the problem doesn't happen, as @mapclyps highlighted.

**Comment by fabiospampinato** [comment]
I've no idea why this is happening, the same function (https://github.com/fabiospampinato/notable/blob/3641862488ae869b41bdbf57f611666f9311f811/src/renderer/containers/main/note.ts#L54) ultimately gets called both when clicking the button and when using the shortcut, but for some reason if it's called from the button's onClick handler then this bug appears. 

I've tried calling the function with a delay, nothing. I've tried refocusing to the editor, nothing. I've tried calling another function with in turns calls `note.new`, this works, and I also have no idea why.

If anyone would like to take a look at this I'd appreciate it, in the meantime I'm going to catch some zzz's.

Ps. there's actually a bug [here](https://github.com/fabiospampinato/notable/blob/3641862488ae869b41bdbf57f611666f9311f811/src/renderer/containers/main/index.ts#L84-L85), by fixing it I can no longer reproduce the problem, but regardless of this bug both clicking the button and using the shortcut should have the same exact effect. Since I can't reproduce this bug anymore if I fix the other (the 2 are unrelated) I haven't pushed the fix to master yet.

**Comment by jinjie** [comment]
@fabiospampinato Tried to look into this matter. I realised that this happens when it tries to toggle to edit mode, if it's already in editing mode. That is why it happens only when it's in toggling mode.

I do think it's a weird behaviour, it should work the same in both shortcut and on the button.

I did a tried a simple fix in PR #325 and it seem to work well.

**Comment by fabiospampinato** [comment]
Found the issue, actually the two actions don't call the same function, as `note.new` is actually patched by [unstated-suspense-autosuspend](https://github.com/fabiospampinato/unstated-suspense-autosuspend), which performs the patching [at the end](https://github.com/fabiospampinato/overstated/blob/4878ec27e7e8205a1fb52903acca9508dc45e862/src/index.ts#L23) of the current call stack, in order to support static properties which are defined by TypeScript only after the constructor has been run.

I'm not sure what's the best way of fixing this, but at least I know what the problem is now.

**Comment by fabiospampinato** [comment]
@alexanderte Can you post some reproducible steps?

The gif attached in the first post highlighted a particular problem, which I can no longer reproduce.

**Comment by alexanderte** [comment]
Of course.

1. Delete all notes
2. Create new note using Note → New
3. Change heading to “# Untitled123”
4. Create new note using Note → New

The new note is now a duplicate. It should be noted that when I repeat these steps, then it leads to different results. Sometimes there is a mix of duplicates and new notes. Hope this helps!

**Comment by dpecos** [comment]
I also noted that the new note "replaces" (being a duplicate of an existing one) "replaces" another one, effectively deleting random notes :(

**Comment by fabiospampinato** [comment]
@dpecos I'm not sure I'm understanding this right, a new note gets created with the content of the previous one, how is this deleting random notes?

**Comment by dpecos** [comment]
These are the (aprox) steps I followed (but is not always happening):
1 - I was viewing (or editing, not sure) a note 
2 - Clicked on the plus button
3 - A "new" note was created with the same content as the one I was viewing
4 - Some note that was there before now is gone (and I think it was the one next to the cloned one, but again, not sure

I've checked the folder to see if the missing file was there, but no, it was gone completely.

From dropbox I was able to recover the missing note.

**Comment by xiety** [comment]
Reproduced every time for me (1.3.0 win10)
- create a new note
- change it's text from Unititled to a
- wait for the list to update it's title
- create a new note, content will stay from a (duplicate bug)
- wait for the list to update it's title from Untitiled to a
- change it's text from a to b
- wait for the list to update it's title
- click on a, the content will stay b, wait and there will be two b-s in the list, replacing the content of the note a (note is lost bug)

**Comment by fabiospampinato** [comment]
Closing in favor of #469 which seems to be the same bug.
